classes/02class/exercises/c02-network07/starkmatt/c02-network07.txt

```diff
@@ -0,0 +1,31 @@
+##Exercise 7
+- VPC Dashboard service, and on the left hand side select Route Tables
+
+- Then select my Public RT, and added the following 0.0.0.0/0 | target devopsacademy-igw igw-080859d0d893c8a00
+
+- After associating, I then selected my private route table added the same details as above but I selected the NAT Gateway as the target | devopsacademy-ngw nat-07d1ac4edb3e79d4b
+
+- After creating RT's, I then went into Subnet Associations and selected Private Subnets for the Private RT, and Public Subnets for the Public RT
+
+Why did you add 0.0.0.0 route to the IGW on public route table?
+0.0.0.0 was used so it covers awhole range of addresses so they're able to access the Public Instance
+As per AWS Networking rules, an Internet Gateway is needed for accessibility from the public network into the AWS Environment.
+
+Why did you add 0.0.0.0 route to the NGW on private route table?
+A Nat Gateway is needed for the private network for accessibility, once the public subnet is accessible you're able to route to the Private subnet.
+
+
+What is the difference of IGW and NGW
+- Internet Gateway gives access public subnet access to the internet.
+- NAT Gateway gives Internet access but under strict rules.
+Those are having a one way connection out but not back in, you're able to use an ACL to control the traffic to and from the subnet in which the NAT Gateway is located
+
+
+Can you delete the destination route to your VPC network? Why?
+- The local VPC that was created initally ?
```

Comment:
It means the default route added that maps to your local VPC. You cannot delete it in AWS.

From https://docs.aws.amazon.com/vpc/latest/userguide/VPC_Route_Tables.html
>Every route table contains a local route for communication within the VPC. This route is added by default to all route tables. If your VPC has more than one IPv4 CIDR block, your route tables contain a local route for each IPv4 CIDR block. If you've associated an IPv6 CIDR block with your VPC, your route tables contain a local route for the IPv6 CIDR block. You cannot modify or delete these routes in a subnet route table or in the main route table.